### PR TITLE
docs(resolver): add authorization example

### PR DESCRIPTION
**Pull request by char0n**
### Description
Add authorization example for OAS resolver

### Motivation and Context

Frequent support questions around this topic

### Types of changes

- [x] No code changes (changes to documentation, CI, metadata, etc)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/usage/openapi-definition-resolver.md [hunk 1]
```diff
@@ -51,6 +51,36 @@ SwaggerClient.resolve({ url: 'https://raw.githubusercontent.com/swagger-api/swag
 > *__Note:__ you can provide `Swagger.resolve` both `spec` and `url` options. In that case
 `spec` will always be preferred and `url` option will be ignored.* 
 
+#### Authentication
+
+When JSON-References points to a remote document protected by authentication,
+`requestInterceptor` option can be used to provide e.g. an authentication header.
```
Comment (tim-lai) [inline docs/usage/openapi-definition-resolver.md:57]: `provide`... what? recommend specifying additional examples

File: docs/usage/openapi-definition-resolver.md [hunk 2]
```diff
@@ -51,6 +51,36 @@ SwaggerClient.resolve({ url: 'https://raw.githubusercontent.com/swagger-api/swag
...
+
+```js
+import SwaggerClient from 'swagger-client';
+
+const requestInterceptor = request => {
+  if (request.url.startsWith('https://raw.githubusercontent.com/swagger-api/')) {
```
Comment (hkosova) [inline docs/usage/openapi-definition-resolver.md:63]: @char0n Would it be better to use `if (request.loadSpec)` here? Or at least mention `loadSpec` somewhere in this section. If I remember correctly, `loadSpec: true` is set any time Swagger JS fetches a document - this includes both the main definition specified by `url` and any external `$refs`.

File: docs/usage/openapi-definition-resolver.md (hunk 1 above)
Comment (char0n) [reply docs/usage/openapi-definition-resolver.md]: Changed it to

> When JSON-References points to a remote document protected by authentication,
`requestInterceptor` option can be used to provide request authentication (e.g. by setting `Authentication` header).

Let me know if it needs some additional tweaking.

File: docs/usage/openapi-definition-resolver.md (hunk 2 above)
Comment (char0n) [reply docs/usage/openapi-definition-resolver.md]: @hkosova right I've overlooked  `loadSpec` property when consolidating new documentation. Checkout the PR now. I've added a full new section related to `loadSpec` and couple of examples. Check it out and let me know If I could change/add something more to make it more understandable to SwaggerClient consumers.

File: docs/usage/http-client.md [hunk 3]
```diff
@@ -84,6 +84,7 @@ Type notations are formatted like so:
 Property | Description
 --- | ---
 `query` | `Object=null`. When provided, HTTP Client serialize it and appends the `queryString` to `Request.url` property.
+`loadSpec` | `Boolean=undefined`. This property will be present and set to `true` when the `Request` was constructed internally by `SwaggerClient` to fetch the OAS definition befined by `url` or when resolving remote JSON References.
```
Comment (hkosova) [inline docs/usage/http-client.md:87]: typo: befined -> defined

Comment (char0n) [inline docs/usage/http-client.md:87]:
```suggestion
`loadSpec` | `Boolean=undefined`. This property will be present and set to `true` when the `Request` was constructed internally by `SwaggerClient` to fetch the OAS definition defined by `url` or when resolving remote JSON References.
```

Comment (char0n) [reply docs/usage/http-client.md]: Fixed in 2e2e47c9f68b45d8e4ec52dc86f7f1b8357fea89

**Comment by char0n** [comment]
Thank you all for comments! Merging.

**Pull request merged**